salt/docker/init.sls

```diff
@@ -29,13 +84,31 @@ include:
 # docker daemon
 #######################
 
-{% set docker_args = salt.caasp_pillar.get('docker:args') %}
-{% set docker_logs = salt.caasp_pillar.get('docker:log_level') %}
-{% set docker_reg  = salt.caasp_pillar.get('docker:registry') %}
-{% set docker_opts = docker_args + " --log-level=" + docker_logs %}
-{% if docker_reg %}
-  {% set docker_opts = docker_opts + " --insecure-registry=" + docker_reg + " --registry-mirror=http://" + docker_reg  %}
-{% endif %}
+{% set docker_daemon_config = salt.caasp_pillar.get('docker:daemon') %}
+{% set registries           = salt.caasp_pillar.get('registries') %}
+
+/etc/docker/daemon.json:
+  file.managed:
+    - contents: |
+{
+  "iptables": "{{ docker_daemon_config.get('iptables') }}",
+  "log_level": "{{ docker_daemon_config.get('log_level') }}",
+  "registries": [
+  {% for registry in registries -%}
+    {% set mirrors = registry.get('mirrors') -%}
```

Comment:
I thing it should be just `"mirror"`

Reply:
I would do something like `{% set is_mirror = registry.get('mirror') -%}`

Reply:
And would it need a default value? Something like `{% set is_mirror = registry.get('mirror', False) -%}`